### PR TITLE
Ers 456 auto add users from ldap to usm

### PR DESCRIPTION
**Pull request by robertkrantz**

### CODE REVIEW

**Comment by robertkrantz** [comment]
test me

**Comment by robertkrantz** [comment]
test me

File: authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java [hunk 1]
```diff
@@ -50,316 +47,287 @@
...
+        }
+    }
+
+    private void handleLoginFailure(String userName) {
+        LOGGER.debug("handleLoginFailure(" + userName + ") - (ENTER)");
+
+        dao.recordLoginFailure(userName);
+
+        Properties props = policyProvider.getProperties(AUTHENTICATION_SUBJECT);
+        int threshold = policyProvider.getIntProperty(props, LOCKOUT_FRESHOLD, 0);
```
Comment (RobinMilding) [inline authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java:360]: lol "freshold"...

File: authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java [hunk 2]
```diff
@@ -326,13 +335,23 @@ private void handleSyncWithLDAP(Map<String, Object> userMap, String userName) {
     }
 
     private boolean personDoesNotExistInDatabase(String username) {
-        return dao.getPersonId(username) != 0;
+        return dao.getPersonId(username) == 0;
```
Comment (RobinMilding) [inline authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java:337]: Oh wow...

**Review by RobinMilding — APPROVED** [review]
Lots of good cleanup. If you want even more to do, there are a lot of calls to log.info() that should really be calls to log.debug() or even log.trace()...
Also the GitHub feature where you can hide white space changes was very handy for these commits :)

**Comment by robertkrantz** [comment]
test me

File: authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java (hunk 1 above)
Comment (robertkrantz) [reply authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java:360]: Would love to change it but I don't know who sets that property so I don't dare touch it.

**Review by RobinMilding — APPROVED** [review]

**Review by ebrygg — APPROVED** [review]

**Pull request merged**